broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -203,6 +208,28 @@ protected void onActorStarted() {
         .registerComponent(zeebePartitionHealth.getName(), zeebePartitionHealth);
   }
 
+  @Override
+  public ActorFuture<Void> closeAsync() {
+    if (closeFuture != null) {
+      return closeFuture;
+    }
+
+    closeFuture = new CompletableActorFuture<>();
+
+    actor.call(
+        () ->
+            // allows to await current transition to avoid concurrent modifications and
+            // transitioning
+            currentTransition()
+                .onComplete(
+                    (nothing, err) -> {
+                      LOG.debug("Closing Zeebe Partition {}.", context.getPartitionId());
+                      super.closeAsync();
+                    }));
```

Comment:
Do we really want to wait until the current transition is done? I'm not sure how long this can take :thinking: 

Since we want to close the partition, it may be faster to interrupt the current transition.

Reply:
This is not safe and not easy to do so. We talking about a graceful shutdown, then I think we should close everything we have opened otherwise we might end up in an inconsistent state, as we did. How long do you think it will take and why? In all our qa tests we doing a normal close and it worked right?

Reply:
My only issue is that this will wait for all the enqueued transitions instead of just the one currently running. During normal operations we probably won't have many transitions but if we do then we'll take a long time to close. One alternative would be to store the "new" futures in a list when we transition instead of replacing the current future with them. We'd only replace the current one when the current finishes. This would then allow us to "jump" the queue when closing it, just wait for the current one and close after that one. However, it adds complexity to the PartitionTransition behavior and it's less elegant than the current solution, so I like it the way it is. Still, I thought it was worth discussing. Thoughts?

Reply:
I discussed this with @saig0 . We decided to keep it that way for now, since it is safe and simple. If we see any issues regarding closing time we can investigate here further how to interrupt or close faster. Personally I think it might be dangerous to do that or at least it needs more effort to do it right.